lib/stat_tracker.rb

```diff
@@ -18,4 +18,287 @@ def self.create_items(file, item_class)
       CSV.foreach(file, csv_options) { |row| item_class.add(item_class.new(row.to_hash)) }
   end
 
+  def total_number_games_by_team_id?(team_id)
+    games = GameTeam.all
+    total_games = 0
+
+    games.each do |game|
+      total_games += 1 if game.last.key?(team_id)
+    end
+    total_games
+  end
+
+  def all_game_teams_by_team_id(team_id)
+    game_teams = GameTeam.all
+    all_game_teams = []
+
+    game_teams.each do |game|
+      all_game_teams << game.last[team_id] if game.last.key?(team_id)
+    end
+    all_game_teams
+  end
+
+  def all_games_by_team_id(team_id)
+    games = Game.all
+    all_games = []
+
+    games.each do |game|
+      if game.last.home_team_id == team_id || game.last.away_team_id == team_id
+        all_games << game.last
+      end
+    end
+    all_games
+  end
+
+  def total_results_by_team_id(team_id)
+    all_game_teams_by_team_id(team_id).map(&:result)
+  end
+
+  def average_win_percentage(team_id)
+    total_games = total_number_games_by_team_id?(team_id).to_f
+    return -0.0 if total_games.zero?
+
+    total_wins = total_results_by_team_id(team_id).map do |game|
+      game == "WIN" ? 1 : nil
+    end.compact.sum
+
+    (total_wins / total_games).round(2)
+  end
+
+  def all_goals_scored_by_team_id(team_id)
+    all_game_teams_by_team_id(team_id).map do |game|
+      game.goals
+    end
+  end
+
+  def score_differences_by_team_id(team_id)
+    games = Game.all
+    goal_differences = []
+
+    games.each do |game|
+      if game.last.home_team_id == team_id
+        goal_differences << (game.last.home_goals - game.last.away_goals)
+      elsif game.last.away_team_id == team_id
+        goal_differences << (game.last.away_goals - game.last.home_goals)
+      end
+    end
+    goal_differences
+  end
+
+  def get_team_name(team_id)
```

Comment:
Helper method for getting team name by team_id

Reply:
I think I have a similar helper method called `return_team_name` not something that has to be changed but just as a note for refactoring